src/state.py

```diff
@@ -253,23 +289,34 @@ def from_charm(  # pylint: disable=too-many-locals
             Current state of the charm.
 
         Raises:
-            InvalidStateError: if the state is invalid.
+            InvalidStateError: if the state is invalid and unrecoverable.
+            RecoverableStateError: if the state is invalid and recoverable.
         """
+        filebeat_username, filebeat_password, endpoints = _fetch_filebeat_configuration(
+            charm.model, indexer_relation_data
+        )
+        args = {key.replace("-", "_"): value for key, value in charm.config.items()}
+        # mypy doesn't like the str to Url casting
```

Comment:
What is the mypy error please?

Reply:
src/state.py:302: error: Argument 1 to "WazuhConfig" has incompatible type "**dict[str, int | float | str]"; expected "Url | None"  [arg-type]

Reply:
I think mypy is more unhappy with the dictionary unpacking in general. The type of `charm.config` is a dictionary with any keys where values van be `int | float | str` whereas the class expects specific keys and types.  It is probably ok in this case because pydantic will check that everything is fine, so perhaps update the comment saying that mypy checks don't add value here as the argument typs are checked by pydantic?